doc/source/analysis.rst

```diff
@@ -1,5 +1,400 @@
 Graph analysis
 ==============
+|igraph| enables analysis of graphs/networks from simple operations such as adding and removing nodes to complex theoretical constructs such as community detection. Read the `API documentation`_ for details on each function and class.
 
-.. note:: TODO. This is a placeholder section; it is not written yet.
+The context for the following examples will be to import the :class:`Graph` class and to have one or more graphs available:
 
+>>> import igraph as ig
+>>> from igraph import Graph
+>>> g = Graph(edges=[[0, 1], [2, 3]])
+
+To get a summary representation of the graph, use :meth:`Graph.summary`. For instance
+
+>>> g.summary(verbosity=1)
+
+will provide a fairly detailed description.
+
+To copy a graph, use :meth:`Graph.copy`.
+
+Vertices and edges
++++++++++++++++++++++++++++
+Vertices are numbered 0 to `n-1`, where n is the number of vertices in the graph. These are called the "vertex ids".
+To count vertices, use :meth:`Graph.vcount`:
+
+>>> n = g.vcount()
+
+Edges also have ids from 0 to `m-1` and are counted by :meth:`Graph.ecount`:
+
+>>> m = g.ecount()
+
+To get a sequence of vertices, use their ids and :attr:`Graph.vs`:
```

Comment:
If we mention IDs, we should illustrate how to use IDs for subsetting `g.vs` (i.e. `g.vs[2]` or `g.vs[2,4,6]`).

Reply:
yes added now